AutonomousDemo/VehicleMotion.cs

```diff
@@ -10,24 +10,34 @@ public class VehicleMotion
     {
         public Vehicle Vehicle;
         public int Time;
+
         public double CurrentSpeed { get; set; }
         public double CurrentAcceleration { get; set; }
         public Velocity CurrentVelocity { get; set; }
         
-        public VehicleMotion(double currentSpeed)
+        public VehicleMotion(double currentSpeed, double meters, int seconds)
```

Comment:
I think some of my confusion might be alleviated if I could see how you intend for the simulation to run. Take a look at http://gameprogrammingpatterns.com/game-loop.html (just the first decent result I got for a search, not a source I am particularly attached to).